src/fio.c

```diff
@@ -217,18 +220,22 @@ void fio_deserialize_state(void) {
     SaveData* saveData = &gCurrentSaveFile;
     s32 i, j;
 
-    gPlayerData = saveData->player;
-
-    gGameStatusPtr->areaID = saveData->areaID;
-    gGameStatusPtr->mapID = saveData->mapID;
-    gGameStatusPtr->entryID = saveData->entryID;
-    gGameStatusPtr->savedPos.x = saveData->savePos.x;
-    gGameStatusPtr->savedPos.y = saveData->savePos.y;
-    gGameStatusPtr->savedPos.z = saveData->savePos.z;
-
-    for (i = 0; i < ARRAY_COUNT(gCurrentEncounter.defeatFlags[0]); i++) {
-        for (j = 0; j < ARRAY_COUNT(gCurrentEncounter.defeatFlags); j++) {
-            gCurrentEncounter.defeatFlags[j][i] = saveData->enemyDefeatFlags[j][i];
+    if (saveData->version[0] == '\0') {
+        deserialize_vanilla_save(saveData);
+    } else {
```

Comment:
Should check the version string matches and warn on console if it doesn't
Modders should be encouraged to have a porting system